src/tests/cmocka/test_dp_request_table.c

```diff
@@ -0,0 +1,338 @@
+/*
+    Copyright (C) 2015 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <talloc.h>
+#include <tevent.h>
+#include <errno.h>
+#include <popt.h>
+
+#include "providers/backend.h"
+#include "providers/data_provider/dp_private.h"
+#include "tests/cmocka/common_mock.h"
+#include "tests/common.h"
+
+struct test_ctx {
+    hash_table_t *table;
+};
+
+static int test_setup(void **state)
+{
+    struct test_ctx *test_ctx = NULL;
+    errno_t ret;
+
+    assert_true(leak_check_setup());
+
+    test_ctx = talloc_zero(global_talloc_context, struct test_ctx);
+    assert_non_null(test_ctx);
+
+    ret = dp_req_table_init(test_ctx, &test_ctx->table);
+    assert_int_equal(ret, EOK);
+    assert_non_null(test_ctx->table);
+
+    *state = test_ctx;
+
+    return 0;
+}
+
+static int test_teardown(void **state)
+{
+    talloc_zfree(*state);
+    assert_true(leak_check_teardown());
+    return 0;
+}
+
+static void test_add_del_req(void **state)
+{
+    errno_t ret;
+    bool is_present;
+    const char *req_name;
+    TALLOC_CTX *mem_ctx = global_talloc_context;
+    struct test_ctx *test_ctx = talloc_get_type(*state, struct test_ctx);
+
+    req_name = dp_req_table_key(mem_ctx,
+                                DPT_ID, DPM_ACCOUNT_HANDLER,
+                                DP_FAST_REPLY,
+                                "custom_part");
+    assert_string_equal(req_name, "0:0:0x0001:custom_part");
+
```

Comment:
Please, do not test the key generator on multiple cases, rather create a separate test for it and then just use it else where. If we change the key string to something else we will have to amend it only on one place.

And you are missing assert_non_null either way.